### PR TITLE
Bug #6721: Don't delete LUNs that are in use

### PR DESCRIPTION
**Pull request by KaustubhDhokte**
Added a warning message while deleting active iscsi target/Associated extent. Removed the ExtentDelete hack.

### CODE REVIEW

File: gui/freeadmin/options.py [hunk 1]
```diff
@@ -73,7 +73,6 @@ class BaseFreeAdmin(object):
     create_modelform = None
     edit_modelform = None
     delete_form = None
-    delete_form_filter = {}  # Ugly workaround for Extent/DeviceExtent
```
Comment (williambr) [inline gui/freeadmin/options.py]: Why did you remove that? Please add it back.

File: gui/services/forms.py [hunk 2]
```diff
@@ -1823,6 +1823,22 @@ def clean_iscsi_target_authgroup(self):
...
+    '''
+    Delete form for Targets/Associated Targets
+    '''
+    def __init__(self, *args, **kwargs):
+        self.instance = kwargs.pop('instance', None)
+        super(TargetExtentDelete, self).__init__(*args, **kwargs)
+        if not self.data:
+            connected_targets = notifier().iscsi_connected_targets()
+            # Get target name from targetname/extentname combination
+            target_to_be_deleted = str(self.instance).split(' / ')[0]
```
Comment (williambr) [inline gui/services/forms.py]: Getting the str representation of self.instance and splitting it looks really weird? How about using direct instance fields?

self.instance.iscsi_whatever

File: gui/services/forms.py [hunk 3]
```diff
@@ -1833,13 +1849,31 @@ class ExtentDelete(Form):
...
+            self.fields.pop('delete')
+        if not self.data:
+            targets_in_use = notifier().iscsi_connected_targets()
+            is_extent_active = False
+            target_to_extent_list = models.iSCSITargetToExtent.objects.filter(
+                iscsi_extent__iscsi_target_extent_name=str(
+                    self.instance).split()[0])
+            for target_to_extent in target_to_extent_list:
+                # Get target name from target:extent association
+                target = str(target_to_extent).split(' / ')[0]
```
Comment (williambr) [inline gui/services/forms.py]: Same here regarding str(target_to_extent)

File: gui/services/forms.py (hunk 2 above)
Comment (KaustubhDhokte) [inline gui/services/forms.py]: This make sense, but as the form is used for both target delete and target-extent delete, few more lines of code needs to be added, 
           ```
 if isinstance(self.instance, models.iSCSITarget):
     target_to_be_deleted = self.instance.iscsi_target_name
 elif isinstance(self.instance, models.iSCSITargetToExtent):
     target_to_be_deleted = self.instance.iscsi_target.iscsi_target_name
```
I have already commented what is being happening there. Do you still want me to make this change (as it add extra lines) or an elaborated comment should do?

This perfectly make sense in case of ExtentDelete.

Comment (williambr) [inline gui/services/forms.py]: I'd prefer if you use the code above. Parsing a string is a really bad practice, not to say it would break with any change to that string representation.

Thanks!

**Comment by williambr** [comment]
Please move your redmine ticket status to Build Testing

Thanks

**Comment by KaustubhDhokte** [comment]
Done. Thank you :)
